bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -466,80 +487,141 @@ public void start() {
-                            LOG.info("Completed checkAllLedgers in {} milliSeconds", checkAllLedgersDuration);
-                            checkAllLedgersTime.registerSuccessfulEvent(checkAllLedgersDuration, TimeUnit.MILLISECONDS);
-                        } catch (KeeperException ke) {
-                            LOG.error("Exception while running periodic check", ke);
-                        } catch (InterruptedException ie) {
-                            Thread.currentThread().interrupt();
-                            LOG.error("Interrupted while running periodic check", ie);
-                        } catch (BKException bke) {
-                            LOG.error("Exception running periodic check", bke);
-                        } catch (IOException ioe) {
-                            LOG.error("I/O exception running periodic check", ioe);
-                        } catch (ReplicationException.UnavailableException ue) {
-                            LOG.error("Underreplication manager unavailable running periodic check", ue);
+            executor.scheduleAtFixedRate(new Runnable() {
+                public void run() {
+                    try {
+                        if (!ledgerUnderreplicationManager.isLedgerReplicationEnabled()) {
+                            LOG.info("Ledger replication disabled, skipping checkAllLedgers");
+                            return;
                         }
+
+                        Stopwatch stopwatch = Stopwatch.createStarted();
+                        LOG.info("Starting checkAllLedgers");
+                        checkAllLedgers();
+                        long checkAllLedgersDuration = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+                        LOG.info("Completed checkAllLedgers in {} milliSeconds", checkAllLedgersDuration);
+                        checkAllLedgersTime.registerSuccessfulEvent(checkAllLedgersDuration, TimeUnit.MILLISECONDS);
+                    } catch (KeeperException ke) {
+                        LOG.error("Exception while running periodic check", ke);
+                    } catch (InterruptedException ie) {
+                        Thread.currentThread().interrupt();
+                        LOG.error("Interrupted while running periodic check", ie);
+                    } catch (BKException bke) {
+                        LOG.error("Exception running periodic check", bke);
+                    } catch (IOException ioe) {
+                        LOG.error("I/O exception running periodic check", ioe);
+                    } catch (ReplicationException.UnavailableException ue) {
+                        LOG.error("Underreplication manager unavailable running periodic check", ue);
                     }
-                    }, initialDelay, interval, TimeUnit.SECONDS);
+                }
+                }, initialDelay, interval, TimeUnit.SECONDS);
+        } else {
+            LOG.info("Periodic checking disabled");
+        }
+    }
```

Comment:
@reddycharan I am guessing above two routines are just refactoring. Let me know if they need a close look for logic change.

Reply:
yes it is simple refactoring, didn't change anything